code/game/machinery/poolcontroller.dm

```diff
@@ -57,33 +57,32 @@
 		updateUsrDialog()
 		return
 
-//herpderp	else
-//herpderp		if(stat & (NOPOWER|BROKEN))
-//herpderp			return
-//herpderp		if (istype(W,/obj/item/weapon/reagent_containers/glass/beaker/large))
-//herpderp			if(src.beaker)
-//herpderp				user << "A beaker is already loaded into the machine."
-//herpderp				return
+	else
+		if(stat & (NOPOWER|BROKEN))
+			return
+		if (istype(W,/obj/item/weapon/reagent_containers/glass/beaker/large))
```

Comment:
still,why check the recipient and not the actual chem? just check if there's just a chem,and if there is,check if its quantity is 100,no?